plugins/com.google.cloud.tools.eclipse.appengine.newproject/src/com/google/cloud/tools/eclipse/appengine/newproject/flex/AppEngineFlexWizardPage.java

```diff
@@ -18,18 +18,41 @@
 
 import com.google.cloud.tools.eclipse.appengine.newproject.AppEngineWizardPage;
 import com.google.cloud.tools.eclipse.appengine.newproject.Messages;
+import com.google.cloud.tools.eclipse.appengine.newproject.maven.MavenCoordinatesUi;
+import org.eclipse.jface.layout.GridLayoutFactory;
 import org.eclipse.swt.widgets.Composite;
 import org.eclipse.swt.widgets.Shell;
 import org.eclipse.ui.PlatformUI;
 
 public class AppEngineFlexWizardPage extends AppEngineWizardPage {
+
+  private MavenCoordinatesUi mavenCoordinatesUi;
+
   public AppEngineFlexWizardPage() {
     super(false);
     setTitle(Messages.getString("app.engine.flex.project")); //$NON-NLS-1$
     setDescription(Messages.getString("create.app.engine.flex.project")); //$NON-NLS-1$
   }
 
   @Override
+  public void createControl(Composite parent) {
+    super.createControl(parent);
+
+    Composite composite = (Composite) getControl();
+    mavenCoordinatesUi = new MavenCoordinatesUi(composite, true /* dynamic enabling */);
+    mavenCoordinatesUi.addChangeListener(new PageValidator());
+    GridLayoutFactory.swtDefaults().generateLayout(composite);
+  }
+
+  @Override
+  public boolean validatePage() {
+    if (!super.validatePage()) {
+      return false;
+    }
+    return mavenCoordinatesUi.setValidationMessage(this);
+  }
+
+  @Override
   public void sendAnalyticsPing(Shell shell) {
     // TODO: send analytics ping
```

Comment:
We have an issue for this: https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1671